hkmc2/shared/src/test/mlscript-compile/TreeTracer.mls

```diff
@@ -0,0 +1,45 @@
+import "./Predef.mls"
+
+open Predef { fold }
+
+module TreeTracer with ...
+
+class TreeTracer with
+  mut val steps = 0
+  mut val enabled = false
+  fun output(outie, innie, innieAlt, message) = if enabled do
+    let lines = message.split("\n")
+    lines.forEach of (line, i, xs) => if
+      let postfix = if lines.length > 1 then " ↵" else ""
+      i is 0 then console.log(outie + innie + line + postfix)
+      i + 1 == lines.length then console.log(outie + innieAlt + line)
+      else console.log(outie + innieAlt + line + postfix)
+  fun enter(...pieces) =
+    output of
+      if steps > 0 then "│ ".repeat(steps - 1) + "├─" else ""
+      if steps > 0 then "┮ " else "┍ "
+      "│ "
+      fold (+) of ...pieces
```

Comment:
Why the space before the first argument list?
```suggestion
      fold(+) of ...pieces
```

Reply:
Remove the space before the first argument list of `fold`